azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/aad/AADAuthenticationProperties.java

```diff
@@ -81,6 +80,42 @@
      */
     private boolean allowTelemetry = true;
 
+    public void setUserGroupWithBool(boolean microsoftGraphApi) {
+        if (microsoftGraphApi) {
+            if (!MicrosoftGraphUserGroupProperties.class.isInstance(this.userGroup)) {
+                final AADGraphUserGroupProperties tempProperties = new AADGraphUserGroupProperties();
```

Comment:
To be honest, I am a little confused with this piece of code, When the switch is on(using the version 2 graph APIs) and current variable userGroup is not the instance of MicrosoftGraphUserGroupProperties, why not just assign a new object of MicrosoftGraphUserGroupProperties to the local variable userGroup? Can you explain it to me? Thank you.

Reply:
@neuqlz 
so the one issue I was running into was if there are override properties (in a property file) in the UserGroup, then I was not able to get these applied. i tried a multitude of various annotations/etc but couldnt get anything to stick. the only way i could get this to work was by initializing the usergroup to one of those two property classes at the beginning, and then if setting the usergroup to the other property in the case that microsoft graph api is to be used, then to check to see if any properties were overridden, and if so, then apply them with such a "weird" function.

ideally, if the override properties would get applied to that property, then wouldn't need to do this.  i even tried separating out the MicrosoftGraphUserGroupProperties to its own file, but still couldnt get the override properties to get applied.  (The test where this comes up is the testUserGroupPropertiesAreOverriden in the AADAuthenticationAutoConfigurationTest.java).

Reply:
For the userGroupProperties, why you need to separate it into two child classes? (AADGraphUserGroupProperties and MicrosoftGraphUserGroupProperties) I think you can just use UserGroupProperties. Your purpose is that when you use v2 graph APIs, the variables in UserGroupProperties should be the v2 version value. So you can just configurate it by application.properties, just like:

azure.activedirectory.user-group.allowed-groups=another_group,third_group
azure.activedirectory.user-group.key=@odata.type
azure.activedirectory.user-group.value=#microsoft.graph.group
azure.activedirectory.user-group.object-id-key=id

Reply:
ok sounds good, will change it to that